### ISSUE TITLE
All trackers get connection failed

### ISSUE THREAD

**Issue by fxcoudert** [transmission/transmission#3285]
Opening a magnet link in Transmission very often leads to no peers being obtained, because none of the trackers can be connected to:

<img width="672" alt="Capture d’écran 2022-06-13 à 17 44 11" src="https://user-images.githubusercontent.com/1980544/173392919-99a9c728-be10-490b-a9e1-ff9203c6c48e.png">

On the same machine, same network, opening the same magnet link with another client will show most of the trackers are actually online and working:

<img width="954" alt="Capture d’écran 2022-06-13 à 17 40 21" src="https://user-images.githubusercontent.com/1980544/173393040-838705f5-6ace-486f-8a2f-8e65b00a798f.png">

**Comment by GaryElshaw** [comment]
I thought this might be symptomatic of me starting and stopping different builds. I've found a restart fixes everything.

**Comment by fxcoudert** [comment]
Restarting the app, even with clean slate / no ongoing transfers, does not fix it. It has been happening for a long while, several months. Usually peers are found by another way (DHT, I think? I am not an expert in torrent at all) so files can transfer at some point, but all trackers are in error.

**Comment by GaryElshaw** [comment]
Sorry, i was meaning a computer restart :-(

**Comment by ckerr** [comment]
@fxcoudert this appears to be a duplicate of #2546?

**Comment by fxcoudert** [comment]
Uh, you're right, sorry. Well there is new information, which is other software can definitely connect from the same computer/network/settings. I have no idea how to debug, any advice is welcome.

**Comment by ckerr** [comment]
Usually I close the new issue in favor of the older one; however this one seems to be more actionable

**Comment by Coeur** [comment]
Can you try, @fxcoudert, to change those two lines:
https://github.com/transmission/transmission/blob/eefcbe7cb2eae0a825be08ed9cd9cea5339fe6ef/CMakeLists.txt#L63-L64

To:
```
set(TR_USER_AGENT_PREFIX "3.01")
set(TR_PEER_ID_PREFIX "TR3001")
```

I had a few torrents unblocked (but not all) by changing my version number: the issue could come from trackers banning us?

**Comment by fxcoudert** [comment]
@Coeur it did not change anything for me.

**Comment by fxcoudert** [comment]
Still occurring at e9a7ddf7f941046715e19b29a167efe17170e12e
I can reproduce with a public torrent and public tracker: https://download.manjaro.org/kde/21.3.2/manjaro-kde-21.3.2-220704-linux515.iso.torrent

On the same machine, with this torrent:
- qBittorrent says it can connect to `udp://tracker.opentrackr.org:1337` and retrieve 83 seeds.
- Transmission says "Connection failed"

**Comment by lpla** [comment]
I had this issue today, again. I tried to restart the app several times, but sadly the error kept showing up in all trackers (but tracker.gbitt.info:80, which retrieved clients that were not connected with, I don't know why).

Instead of restarting the computer as I usually do, I tried to modify Transmission settings and restart the app, doing it one by one to find anything that could fix it. The first thing I changed was the listening port to a random one.

That worked.

**Comment by TheFalcon81** [comment]
I am also having this problem with many of the recent v4 for mac nightly builds. very annoying. Not an issue with other BT clients. Never had this issue on version 3.0 on my old intel mac.

**Comment by lluisd** [comment]
I also have that problem now, it was working fine some weeks ago. I solved it by not using Transmission from docker and use the Synology community package. 

I get this errors (not solved with docker and container restarts)
![image](https://user-images.githubusercontent.com/7629843/214264293-6d21b459-2b4d-4620-bc93-4e211425df1b.png)

**Comment by fxcoudert** [comment]
I still get this with Transmission 4.0.0-beta.3 (634b1e8fc1). It says `help wanted` and I am willing to spend time and effort in debugging this, but I have reported all the logs and have no idea how to investigate further. I need guidance, please. @ckerr

**Comment by ckerr** [comment]
@fxcoudert I'm not really sure what to try next. Both of the torrents you shared earlier are working for me. Also from your log messages above it looks like DHT is working for you, and DHT uses UDP for package transport. So it's not as if UDP is firewalled. (Also if it were, other Torrent clients wouldn't work either).

OK, I am just guessing here on how I would start tracking this down this if I were on a system where I could reproduce the problem. Since we know UDP packets are working, but announcing via UDP isn't, my guess is that either DNS lookup is failing or we're not able to send a connect request to the tracker. Let's find out if either of those are the case, here are some tracers:

https://gist.githubusercontent.com/ckerr/8fa07b82a99b70e2866887dd69fb3195/raw/ed0287e11c9397650cee8724d833088411d54937/tracers.diff

Try building with that and we'll see what announcer-udp is actually doing. 

Also if it _is_ a DNS failure, try running `nslookup` from the command line for that server's name.

**Comment by dantearmok** [comment]
Please see #5146

What can we provide to help nail down this issue?

**Comment by fxcoudert** [comment]
I've had this problem consistently for months, on many different builds, but at some point it disappeared for me. This is frustrating, cause I could never identify the cause. I could confirm early on that it was not a DNS failure. But when I came in January to apply to patch provided, things were working (and have been working since).

**Comment by aselafernando** [comment]
Can confirm issue still exists on Debian 11 x64 with Transmission 4.0.3

I added into /etc/sysctl.conf to help increase memory on the TCP/IP buffers
```
net.core.rmem_max=4194304
net.core.wmem_max=1048576
```

Tracker udp://9.rarbg.me:2720 is confirmed to be working with another client (Deluge)

```
[2023-04-18 23:14:41.601] inf session.cc:646 Transmission version 4.0.3 (6b0e49bbb2) starting (session.cc:646)
[2023-04-18 23:14:41.601] dbg cache.cc:129 Maximum cache size set to 4.00 MiB (256 blocks) (cache.cc:129)
[2023-04-18 23:14:41.601] dbg net.cc:328 Bound socket 14 to port XXXX on 0.0.0.0 (net.cc:328)
[2023-04-18 23:14:41.601] inf session.cc:404 Listening to incoming peer connections on [0.0.0.0]:XXXX (session.cc:404)
[2023-04-18 23:14:41.601] dbg net.cc:328 Bound socket 15 to port XXXX on :: (net.cc:328)
[2023-04-18 23:14:41.601] inf session.cc:404 Listening to incoming peer connections on [::]:XXXX (session.cc:404)
[2023-04-18 23:14:41.601] inf tr-udp.cc:168 Bound UDP IPv4 address [0.0.0.0]:XXXX (tr-udp.cc:168)
[2023-04-18 23:14:41.601] inf tr-udp.cc:202 Bound UDP IPv6 address [::]:XXXX (tr-udp.cc:202)
[2023-04-18 23:14:41.601] dbg tr-lpd.cc:281 Initialising Local Peer Discovery (tr-lpd.cc:281)
[2023-04-18 23:14:41.601] dbg tr-lpd.cc:384 Local Peer Discovery initialised (tr-lpd.cc:384)
[2023-04-18 23:14:41.601] dbg tr-dht.cc:139 Starting DHT on port XXXX (tr-dht.cc:139)
[2023-04-18 23:14:41.601] dbg rpc-server.cc:836 setting password-enabled to 'false' (rpc-server.cc:836)
[2023-04-18 23:14:41.601] inf rpc-server.cc:763 Added '127.0.0.1' to host whitelist (rpc-server.cc:763)
[2023-04-18 23:14:41.601] inf rpc-server.cc:763 Added '::1' to host whitelist (rpc-server.cc:763)
[2023-04-18 23:14:41.601] dbg rpc-server.cc:822 setting our username to '' (rpc-server.cc:822)
[2023-04-18 23:14:41.601] dbg rpc-server.cc:830 setting our salted password to 'XXXXXXXXXXXXXXXX' (rpc-server.cc:830)
[2023-04-18 23:14:41.601] inf rpc-server.cc:907 Serving RPC and Web requests on 0.0.0.0:9091/transmission/ (rpc-server.cc:907)
[2023-04-18 23:14:41.601] inf rpc-server.cc:713 Listening for RPC and Web requests on '0.0.0.0:9091' (rpc-server.cc:713)
[2023-04-18 23:14:41.601] inf rpc-server.cc:912 Whitelist enabled (rpc-server.cc:912)
[2023-04-18 23:14:41.601] inf rpc-server.cc:923 Serving RPC and Web requests from '/usr/local/share/transmission/public_html' (rpc-server.cc:923)
[2023-04-18 23:14:41.601] dbg tr-lpd.cc:281 Initialising Local Peer Discovery (tr-lpd.cc:281)
[2023-04-18 23:14:41.601] dbg tr-lpd.cc:384 Local Peer Discovery initialised (tr-lpd.cc:384)
[2023-04-18 23:14:41.601] inf daemon.cc:715 Loading settings from '/var/lib/transmission/.config/transmission-daemon' (daemon.cc:715)
[2023-04-18 23:14:41.601] dbg XXXX Read resume file '/var/lib/transmission/.config/transmission-daemon/resume/XXXX.resume' (resume.cc:650)
[2023-04-18 23:14:41.601] dbg XXXX Read resume file '/var/lib/transmission/.config/transmission-daemon/resume/XXXX.resume' (resume.cc:650)
[2023-04-18 23:14:41.601] dbg XXXX Read resume file '/var/lib/transmission/.config/transmission-daemon/resume/XXXX.resume' (resume.cc:650)
[2023-04-18 23:14:41.601] dbg XXXX Read resume file '/var/lib/transmission/.config/transmission-daemon/resume/XXXX.resume' (resume.cc:650)
[2023-04-18 23:14:41.601] dbg XXXX Read resume file '/var/lib/transmission/.config/transmission-daemon/resume/XXXX.resume' (resume.cc:650)
[2023-04-18 23:14:41.601] dbg XXXX Read resume file '/var/lib/transmission/.config/transmission-daemon/resume/XXXX.resume' (resume.cc:650)
[2023-04-18 23:14:41.601] dbg XXXX Read resume file '/var/lib/transmission/.config/transmission-daemon/resume/XXXX.resume' (resume.cc:650)
[2023-04-18 23:14:41.601] dbg XXXX Read resume file '/var/lib/transmission/.config/transmission-daemon/resume/XXXX.resume' (resume.cc:650)
[2023-04-18 23:14:41.601] dbg XXXX Read resume file '/var/lib/transmission/.config/transmission-daemon/resume/XXXX.resume' (resume.cc:650)
[2023-04-18 23:14:41.601] dbg XXXX Read resume file '/var/lib/transmission/.config/transmission-daemon/resume/XXXX.resume' (resume.cc:650)
[2023-04-18 23:14:41.601] inf session.cc:1406 Loaded 10 torrents (session.cc:1406)
[2023-04-18 23:14:41.601] dbg tracker.thinelephant.org:12760 DNS lookup succeeded (announcer-udp.cc:435)
[2023-04-18 23:14:41.601] dbg tracker.thinelephant.org:12800 DNS lookup succeeded (announcer-udp.cc:435)
[2023-04-18 23:14:41.601] dbg tracker.tallpenguin.org:15740 DNS lookup succeeded (announcer-udp.cc:435)
[2023-04-18 23:14:41.601] dbg tracker.thinelephant.org:12770 DNS lookup succeeded (announcer-udp.cc:435)
[2023-04-18 23:14:41.601] dbg tracker.tallpenguin.org:15720 DNS lookup succeeded (announcer-udp.cc:435)
[2023-04-18 23:14:41.601] dbg tracker.tallpenguin.org:15730 DNS lookup succeeded (announcer-udp.cc:435)
[2023-04-18 23:14:41.601] dbg tracker.tallpenguin.org:15710 DNS lookup succeeded (announcer-udp.cc:435)
[2023-04-18 23:14:41.601] dbg tracker.slowcheetah.org:14770 DNS lookup succeeded (announcer-udp.cc:435)
[2023-04-18 23:14:41.601] dbg tracker.tallpenguin.org:15750 DNS lookup succeeded (announcer-udp.cc:435)
[2023-04-18 23:14:41.601] dbg tracker.tallpenguin.org:15770 DNS lookup succeeded (announcer-udp.cc:435)
[2023-04-18 23:14:41.601] dbg tracker.thinelephant.org:12720 DNS lookup succeeded (announcer-udp.cc:435)
[2023-04-18 23:14:41.601] dbg 9.rarbg.me:2720 DNS lookup succeeded (announcer-udp.cc:435)
[2023-04-18 23:14:41.601] dbg tracker.slowcheetah.org:14740 DNS lookup succeeded (announcer-udp.cc:435)
[2023-04-18 23:14:41.601] dbg tracker.fatkhoala.org:13770 DNS lookup succeeded (announcer-udp.cc:435)
[2023-04-18 23:14:41.601] dbg tracker.thinelephant.org:12740 DNS lookup succeeded (announcer-udp.cc:435)
[2023-04-18 23:14:42.607] dbg tracker.slowcheetah.org:14770 Got a new connection ID from tracker: XXXX (announcer-udp.cc:323)
[2023-04-18 23:14:42.607] dbg tracker.slowcheetah.org:14770 sending req 0x7f14900c8db0 (announcer-udp.cc:520)
[2023-04-18 23:14:42.607] dbg tracker.slowcheetah.org:14770 sending request w/connection id XXXX (announcer-udp.cc:537)
[2023-04-18 23:14:42.607] dbg tracker.thinelephant.org:12800 Got a new connection ID from tracker: XXXX (announcer-udp.cc:323)
[2023-04-18 23:14:42.607] dbg tracker.thinelephant.org:12800 sending req 0x7f14900bdf30 (announcer-udp.cc:520)
[2023-04-18 23:14:42.607] dbg tracker.thinelephant.org:12800 sending request w/connection id XXXX (announcer-udp.cc:537)
[2023-04-18 23:14:42.607] dbg tracker.thinelephant.org:12760 Got a new connection ID from tracker: XXXX (announcer-udp.cc:323)
[2023-04-18 23:14:42.607] dbg tracker.thinelephant.org:12760 sending req 0x7f14900c1470 (announcer-udp.cc:520)
[2023-04-18 23:14:42.607] dbg tracker.thinelephant.org:12760 sending request w/connection id XXXX (announcer-udp.cc:537)
[2023-04-18 23:14:42.607] dbg tracker.thinelephant.org:12770 Got a new connection ID from tracker: XXXX (announcer-udp.cc:323)
[2023-04-18 23:14:42.607] dbg tracker.thinelephant.org:12770 sending req 0x7f14900c6540 (announcer-udp.cc:520)
[2023-04-18 23:14:42.607] dbg tracker.thinelephant.org:12770 sending request w/connection id XXXX (announcer-udp.cc:537)
[2023-04-18 23:14:42.607] dbg XXXX at tracker.slowcheetah.org:14770 peer counts: 3 seeders, 4 leechers. (announcer.cc:690)
[2023-04-18 23:14:42.607] dbg XXXX at tracker.thinelephant.org:12800 peer counts: 12 seeders, 17 leechers. (announcer.cc:690)
[2023-04-18 23:14:42.607] dbg XXXX at tracker.thinelephant.org:12760 peer counts: 1 seeders, 2 leechers. (announcer.cc:690)
[2023-04-18 23:14:42.607] dbg XXXX at tracker.thinelephant.org:12770 peer counts: 5 seeders, 6 leechers. (announcer.cc:690)
[2023-04-18 23:14:42.607] dbg tracker.slowcheetah.org:14740 Got a new connection ID from tracker: XXXX (announcer-udp.cc:323)
[2023-04-18 23:14:42.607] dbg tracker.slowcheetah.org:14740 sending req 0x7f14900d1cf0 (announcer-udp.cc:520)
[2023-04-18 23:14:42.607] dbg tracker.slowcheetah.org:14740 sending request w/connection id XXXX (announcer-udp.cc:537)
[2023-04-18 23:14:42.607] dbg tracker.thinelephant.org:12720 Got a new connection ID from tracker: XXXX (announcer-udp.cc:323)
[2023-04-18 23:14:42.607] dbg tracker.thinelephant.org:12720 sending req 0x7f14900d6820 (announcer-udp.cc:520)
[2023-04-18 23:14:42.607] dbg tracker.thinelephant.org:12720 sending request w/connection id XXXX (announcer-udp.cc:537)
[2023-04-18 23:14:42.607] dbg tracker.thinelephant.org:12740 Got a new connection ID from tracker: XXXX (announcer-udp.cc:323)
[2023-04-18 23:14:42.607] dbg tracker.thinelephant.org:12740 sending req 0x7f14900cf4d0 (announcer-udp.cc:520)
[2023-04-18 23:14:42.607] dbg tracker.thinelephant.org:12740 sending request w/connection id XXXX (announcer-udp.cc:537)
[2023-04-18 23:14:42.607] dbg XXXX at tracker.slowcheetah.org:14740 peer counts: 13 seeders, 26 leechers. (announcer.cc:690)
[2023-04-18 23:14:42.607] dbg XXXX at tracker.thinelephant.org:12720 peer counts: 4 seeders, 155 leechers. (announcer.cc:690)
[2023-04-18 23:14:42.607] dbg XXXX at tracker.thinelephant.org:12740 peer counts: 8 seeders, 21 leechers. (announcer.cc:690)
[2023-04-18 23:15:10.599] dbg XXXX at tracker.tallpenguin.org:15720 Tracker 'tracker.tallpenguin.org:15720' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:10.599] dbg XXXX at tracker.tallpenguin.org:15710 Tracker 'tracker.tallpenguin.org:15710' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:10.599] dbg XXXX at tracker.tallpenguin.org:15740 Tracker 'tracker.tallpenguin.org:15740' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:10.599] dbg XXXX at tracker.tallpenguin.org:15730 Tracker 'tracker.tallpenguin.org:15730' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:10.599] dbg XXXX at tracker.fatkhoala.org:13770 Tracker 'tracker.fatkhoala.org:13770' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:10.599] dbg XXXX at tracker.tallpenguin.org:15750 Tracker 'tracker.tallpenguin.org:15750' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:10.599] dbg XXXX at 9.rarbg.me:2720 Tracker '9.rarbg.me:2720' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:10.599] dbg XXXX at tracker.tallpenguin.org:15770 Tracker 'tracker.tallpenguin.org:15770' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:11.605] dbg XXXX at tracker.trackerfix.com:80 Tracker 'tracker.trackerfix.com:80' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:11.605] dbg XXXX at tracker.trackerfix.com:80 Tracker 'tracker.trackerfix.com:80' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:11.605] dbg XXXX at tracker.trackerfix.com:80 Tracker 'tracker.trackerfix.com:80' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:11.605] dbg XXXX at tracker.trackerfix.com:80 Tracker 'tracker.trackerfix.com:80' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:11.605] dbg XXXX at tracker.trackerfix.com:80 Tracker 'tracker.trackerfix.com:80' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:11.605] dbg XXXX at tracker.trackerfix.com:80 Tracker 'tracker.trackerfix.com:80' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:11.605] dbg XXXX at tracker.trackerfix.com:80 Tracker 'tracker.trackerfix.com:80' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:11.605] dbg XXXX at tracker.trackerfix.com:80 Tracker 'tracker.trackerfix.com:80' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:11.605] dbg XXXX at tracker.trackerfix.com:80 Tracker 'tracker.trackerfix.com:80' scrape error: Could not connect to tracker (Retrying in 20 seconds) (announcer.cc:1254)
[2023-04-18 23:15:28.596] dbg 9.rarbg.me:2720 Connection failed (announcer-udp.cc:328)
[2023-04-18 23:15:28.596] WRN XXXX at 9.rarbg.me:2720 Announce error: Connection failed (Retrying in 327 seconds) (udp://9.rarbg.me:2720) (announcer.cc:885)
```

**Comment by beroset** [comment]
I'm on version 4.0.3 on Fedora 38 and encountered this problem again today.  Restarting Transmission does not fix it; logging out and back in does not fix it; rebooting seems to fix it.

**Comment by yy-hsu** [comment]
I am facing the same issue, version 4.0.3 on Arch Linux, and cannot fix it by downgrading/restart/reboot.
Every torrent shows the same warning `Scrape error: Could not connect to tracker (announcer.c:1392)`.
I suppose it would be a DNS issue, or some other thing related to network, however I cannot find a way to debug.
Even the debug level shows limited information.

**Comment by al3xtjames** [comment]
I ran into the same issue when I had another client running on the same network with the same port. Changing the port on one of the clients resolved the issue.

**Comment by cmdshft** [comment]
Having this issue right now
![image](https://github.com/transmission/transmission/assets/8571823/03454b8f-fe3e-4e2d-b997-67f636caee0c)

**Comment by adam-bauernfeind** [comment]
Hi!
I have the same issue: using transmission with HTTP trackers always ends up with connection error. What new I have: if I close the app, down-up my wired connection, then start it again, the problem goes away for a time. Simply restarting the app does not work me as well. The tracker is always reachable and I can repeat this as many times as I want without restarting my machine and the same things happens every time, it's deterministic. I think its not a DNS cache problem, because the tracker's IP address never changed (it's a fix IP).

**Comment by tgildea** [comment]
al3xtjames' solution worked for me-- I had another application listening on the port Transmission was trying to use. That could explain why a reboot is fixing it for some. Not to say that someone else isn't experiencing a resource leak of some kind, but in my case it was a simple port conflict.

**Comment by twistedpixel** [comment]
Not sure if this is completely related but while downloading, I quite often will notice that after several hours I am connected to perhaps “3 of 7 peers” and the download rate is at a crawl. Trying to update the trackers does nothing or fails but simply closing Transmission and reopening it results in a huge boost in peers. I’m guessing the trackers update correctly at that point and suddenly I’m connected to 20 of 50 peers” or whatever - in any case: significantly more than before.

This is consistent and repeatable in my case.

**Comment by m4dm4x1337** [comment]
Hi,

I can definitely confirm that this is a bug. It only occurs if:

1.) bind-address-ipv4 is 0.0.0.0 or 127.0.0.1
2.) a http/https tracker is connected (tcp connection)

Solution: Use the IP address associated with your default interface as bind-address-ipv4. For example, if your traffic goes through eth0, and eth0 has the IP address 192.168.1.100, then this IP address should be used as bind-address-ipv4.

What happens under the hood is this:

```
root@debian-rpi64:~# LANG=C curl -4ksv --interface 127.0.0.1 http://tracker.openbittorrent.com:80/announce
*   Trying 193.189.100.188:80...
* Name '127.0.0.1' family 2 resolved to '127.0.0.1' family 2
* Local port: 0
* Immediate connect fail for 193.189.100.188: Invalid argument
* Failed to connect to tracker.openbittorrent.com port 80 after 15 ms: Couldn't connect to server
* Closing connection 0
```

The transmission daemon code needs to be modified so that the socket is bound to the default interface when connecting a tracker over tcp.

**Comment by Coeur** [comment]
Hum, 0.0.0.0 is the default:
https://github.com/transmission/transmission/blob/c697d95ad352be74638169fac8280bada8b4c54b/libtransmission/session-settings.h#L23

**Comment by m4dm4x1337** [comment]
I have updated my previous comment and added this line: 2.) bind-address-ipv6 is fe80::

fe80:: is often used to disable ipv6, it's still a bug, because setting bind-address-ipv6 to fe80:: should not affect ipv4 connections, but it does affect it.

There is no bug if bind-address-ipv6 has the default value ::

**Comment by LunarN0v4** [comment]
This is still happening on the latest version of Transmission Qt (openbittorrent is dead, but opentrackr isn't).
![image](https://github.com/transmission/transmission/assets/71067215/cf613ab9-248d-4db9-aa5c-18ceb24405ee)

**Comment by zachminot** [comment]
@LunarN0v4 Open Command Prompt and run netstat -ab, take a look and see if your Transmission listening port is being used by another program, if it is, change the port and try again.

**Comment by mtraynor7** [comment]
> I can definitely confirm that this is a bug. It only occurs if:
> 
> 1.) bind-address-ipv4 is 0.0.0.0 or 127.0.0.1 2.) bind-address-ipv6 is fe80:: 3.) a http/https tracker is connected (tcp connection)
> 
> ...
>
> There is no bug if bind-address-ipv6 has the default value ::

@m4dm4x1337 Are you saying this issue occurs if ALL or ANY of your points are true? I am still experiencing this issue with http trackers but my IP bindings are not set as you describe.

settings.json:

```
"bind-address-ipv4": "10.96.0.38",
"bind-address-ipv6": "::",
```

transmission.log:

```
[2024-01-04 04:48:01.082] WRN [torrent name] at tracker.trackerfix.com:80 Couldn't parse scrape response: no bencoded data to parse (84) (announcer-http.cc:662)
[2024-01-04 05:18:01.080] WRN [torrent name] at tracker.trackerfix.com:80 Couldn't parse scrape response: no bencoded data to parse (84) (announcer-http.cc:662)
[2024-01-04 05:48:01.082] WRN [torrent name] at tracker.trackerfix.com:80 Couldn't parse scrape response: no bencoded data to parse (84) (announcer-http.cc:662)
[2024-01-04 06:18:01.080] WRN [torrent name] at tracker.trackerfix.com:80 Couldn't parse scrape response: no bencoded data to parse (84) (announcer-http.cc:662)
[2024-01-04 06:48:01.086] WRN [torrent name] at tracker.trackerfix.com:80 Couldn't parse scrape response: no bencoded data to parse (84) (announcer-http.cc:662)
[2024-01-04 07:18:01.077] WRN [torrent name] at tracker.trackerfix.com:80 Couldn't parse scrape response: no bencoded data to parse (84) (announcer-http.cc:662)
```

**Comment by tearfur** [comment]
> fe80:: is used to disable IPv6 and that somehow causes TCP4 connections to an http tracker to be bound to the loopback interface.

@m4dm4x1337 You are literally telling Transmission to bind to the loopback interface when you specify `"bind-address-ipv4: "127.0.0.1"`, so I'm confused what you are on about.

It sounds like you've got both IPv4 and IPv6 public Internet, so if you set `"bind-address-ipv6": "fe80::"` and `"bind-address-ipv4": "0.0.0.0"`, you should find that the IPv4 announce succeeds, but IPv6 announce and scrape fails.

**Comment by tearfur** [comment]
I'd like to make it clear that I understand the need for some people to disable IPv6, and though we do not have any official way to do that right now, I'm not arguing against it either.

> I recommend testing it yourself. Use the values "bind-address-ipv6": "fe80::" and "bind-address-ipv4": "0.0.0.0" and then add HTTP trackers to the tracker list of your torrent

Yes I did test on 8e7fc7693021ce279ac4b0cce764cd6e78fb2b25 before writing https://github.com/transmission/transmission/issues/3285#issuecomment-1888412210. Like I said, IPv4 announce will succeed, while IPv6 announce and scrape will fail.

> 1.) bind-address-ipv4 is 0.0.0.0 or 127.0.0.1

I reproduced your IPv4 `Invalid Argument` curl log output only when setting `"bind-address-ipv4": "127.0.0.1"`. It *will be* a bug if `"bind-address-ipv4": "0.0.0.0"` produced the same behaviour, but that was not the case in my tests. 

I'm very curious why you've suggested binding to `127.0.0.1` in the first place, of course that's not gonna work.

> It is absolutely possible and permissible that IPv4 and IPv6 are bound to different interfaces.

Absolutely, and I think we have that currently.

> I could bet that there is a code snippet in the transmission that causes explicit binding for IPv4 connections based on (wrong) IPv6 properties.

To my best knowledge, if there is a bug, you'll be able to find it if you start around here and trace upwards in the call stack:

https://github.com/transmission/transmission/blob/1de6e93fec61b88c9422d4fdc011b7ae2dc59130/libtransmission/web.cc#L309

https://github.com/transmission/transmission/blob/1de6e93fec61b88c9422d4fdc011b7ae2dc59130/libtransmission/web.cc#L322

Please feel free to take a look.

**Comment by arbv** [comment]
I have the same issue (running 4.0.6 from NixOS repos).

**Comment by arbv** [comment]
> I have the same issue (running 4.0.6 from NixOS repos).

In my case it was caused by the fact that Transmission could not access `/run/systemd/resolve` (I use `systemd-resolved`) due to `systemd` unit hardening. See here: https://github.com/NixOS/nixpkgs/issues/98904#issuecomment-716656576

Something similar might be happening on other distribution. If you experience this - check the contents of your Tranmission's `systemd`, AppArmor profiles and similar security-related things.

**Comment by radicalaction** [comment]
![tsm](https://github.com/transmission/transmission/assets/62557479/4193e228-bc61-414c-9bc0-16389f6e9f27)
I restarted my machine, so downloads started. But the error messages didn't go away

**Comment by Coeur** [comment]
@radicalaction An HTTP code starting with a 5, like 521, means it's an issue on the server/tracker side, not the client side.

**Comment by radicalaction** [comment]
> @radicalaction An HTTP code starting with a 5, like 521, means it's an issue on the server/tracker side, not the client side.

Seems that you're right. Thanks!

**Comment by Hazpartame** [comment]
Hello, 

I'm in the same situation here.

**Comment by matiasw** [comment]
I am also affected by this issue. With the latest `4.1.0-dev (ed2c6c4085)` version from Git, I get "IPv4 connection failed" for every tracker. Selecting "Show more details" gives "Got a scrape error 'Could not connect to tracker'".

**Comment by d03j** [comment]
same here using 4.0.6 (38c164933e) in a container (lscr.io/linuxserver/transmission:latest).

I can resolve dns (e.g. ping github.com) from inside the container without any problems.

setting the container to a static IP address and "bind-address-ipv4" to that address does nothing.

**Comment by Luigin-01** [comment]
I had the same issue. I'm using Transmission-qt v4.0.6 (38c164933e) on a PC with Windows 11. I opened settings.json located in C:\Windows\ServiceProfiles\LocalService\AppData\Local\transmission-daemon. There I changed  "bind-address-ipv4": from "0.0.0.0", to "MyIPAddress". Then restarted the transmission daemon and EUREKA!!!!! Everything is working like a charm

**Comment by d03j** [comment]
> same here using 4.0.6 ([38c1649](https://github.com/transmission/transmission/commit/38c164933e9f77c110b48fe745861c3b98e3d83e)) in a container (lscr.io/linuxserver/transmission:latest).
> 
> I can resolve dns (e.g. ping github.com) from inside the container without any problems.
> 
> setting the container to a static IP address and "bind-address-ipv4" to that address does nothing.

Nevermind, the issue was the way I had my container set up.  When I had it in a network I use for a few containers, including flexget, it couldn't connect to udp trackers but If I create the container using the default podman network (no --net ...) in podman 5.2.3 it works flawlessly,

**Comment by angelgarrido** [comment]
> I had the same issue. I'm using Transmission-qt v4.0.6 ([38c1649](https://github.com/transmission/transmission/commit/38c164933e9f77c110b48fe745861c3b98e3d83e)) on a PC with Windows 11. I opened settings.json located in C:\Windows\ServiceProfiles\LocalService\AppData\Local\transmission-daemon. There I changed "bind-address-ipv4": from "0.0.0.0", to "MyIPAddress". Then restarted the transmission daemon and EUREKA!!!!! Everything is working like a charm

Using docker image [linuxserver/transmission](https://github.com/linuxserver/docker-transmission) Version [4.0.6-r0-ls272](https://github.com/linuxserver/docker-transmission/releases/tag/4.0.6-r0-ls272)

Exactly the same happened to me, changing binding address to the specific docker ip assigned address, and trackers connections started working.

**Comment by Neustradamus** [comment]
Any progress on it?

**Comment by comesuccingfuccsloot** [comment]
i don't have "bind-address-ipv4" in my settings.json.

instead, i did 2 things that fixed it:
1. completely exit transmission, open task manager, go to processes tab, search for "transmission", end all processes related to transmission (i had 1 after exiting the app)
2. go to my router settings and enabled "ipv6 router advertisement"

dunno which step actually fixed it, maybe just one, maybe both. either way, now i can download.
